cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1316,19 +1328,43 @@ public void subscribe() {
   }
 
   private void subscribe(boolean newSubscription) {
-    subscribe(newSubscription, null, getCurrentActivity());
+    if (getIabTcfMode() != null && getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
+      this.waitForSubscribeConsent(() -> {
+        subscribe(newSubscription, null, getCurrentActivity());
+      });
+    } else {
+      subscribe(newSubscription, null, getCurrentActivity());
+    }
   }
 
   public void subscribe(SubscribedCallbackListener subscribedCallbackListener) {
-    subscribe(false, subscribedCallbackListener, getCurrentActivity());
+    if (getIabTcfMode() != null && getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
+      this.waitForSubscribeConsent(() -> {
+        subscribe(false, subscribedCallbackListener, getCurrentActivity());
+      });
+    } else {
+      subscribe(false, subscribedCallbackListener, getCurrentActivity());
+    }
   }
 
   public void subscribe(SubscribedCallbackListener subscribedCallbackListener, Activity dialogActivity) {
-    subscribe(false, subscribedCallbackListener, dialogActivity);
+    if (getIabTcfMode() != null && getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
+      this.waitForSubscribeConsent(() -> {
```

Comment:
Not required here. Just implement in the main subscribe() function (https://github.com/cleverpush/cleverpush-android-sdk/pull/166/files#diff-dadb76999f5ed7d5f6c363d011ef4b354183a8b5cee4ecd118f2152c882891f1R1370)